diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0c4159..a3d1ae3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,8 +18,6 @@ set(ENABLE_ALLOSERVER_BINOCULARS ON CACHE BOOL "")
 set(ENABLE_RENDERINGPLUGIN_BINOCULARS ON CACHE BOOL "")
 set(ENABLE_UNITYSCRIPTS_BINOCULARS ON CACHE BOOL "")
 set(ENABLE_ALLOUNITYPLAYER ON CACHE BOOL "")
-set(ENABLE_HELICOPTERCONTROL ON CACHE BOOL "")
-set(ENABLE_HELICOPTERCONTROL ON CACHE BOOL "")
 set(ENABLE_UNITYSCRIPTS_HELICOPTERCONTROL ON CACHE BOOL "")
 set(ENABLE_HELICOPTERCONTROL_OSCPLUGIN ON CACHE BOOL "")
 
@@ -124,12 +122,6 @@ endif()
 if(ENABLE_ALLOUNITYPLAYER)
 	add_subdirectory(AlloUnityPlayer)
 endif()
-if(ENABLE_HELICOPTERCONTROL)
-	add_subdirectory(HelicopterControl)
-endif()
-if(ENABLE_HELICOPTERCONTROL)
-	add_subdirectory(HelicopterControl)
-endif()
 if(ENABLE_UNITYSCRIPTS_HELICOPTERCONTROL)
 	add_subdirectory(UnityScripts_HelicopterControl)
 endif()
diff --git a/HelicopterControl_OSCPlugin/ControlPlugin.cpp b/HelicopterControl_OSCPlugin/ControlPlugin.cpp
index 2a952ab..d3790f5 100644
--- a/HelicopterControl_OSCPlugin/ControlPlugin.cpp
+++ b/HelicopterControl_OSCPlugin/ControlPlugin.cpp
@@ -605,11 +605,11 @@ extern "C" void EXPORT_API oscStart()
 	/*
      * Start OSC client to receive phone sensor orientation
      */
-	OrientationPacketListener listener;
+	/*OrientationPacketListener listener;
     s = new UdpListeningReceiveSocket(IpEndpointName( IpEndpointName::ANY_ADDRESS, PORT ), &listener );
     
     s->Run();
-	sBreakBarrier.wait();
+	sBreakBarrier.wait();*/
 }
 //OSC END
 
